runtime/tests/checker/attachments_test.go

```diff
@@ -3831,6 +3831,22 @@ func TestCheckAttachmentsExternalMutation(t *testing.T) {
 
 		errs := RequireCheckerErrors(t, err, 1)
 		assert.IsType(t, &sema.InvalidAccessError{}, errs[0])
+		require.Equal(
+			t,
+			errs[0].(*sema.InvalidAccessError).RestrictingAccess,
+			sema.NewEntitlementSetAccess(
+				[]*sema.EntitlementType{
+					sema.InsertEntitlement,
+					sema.MutateEntitlement,
+				},
+				sema.Disjunction,
+			),
+		)
+		require.Equal(
+			t,
+			errs[0].(*sema.InvalidAccessError).PossessedAccess,
+			sema.UnauthorizedAccess,
+		)
```

Comment:
Require the type to be correct, then cast and assert facts about parts of the errors individually, so a failure of either part is reported individually (avoids error reporting fatigue, e.g. when the first part-requirement is fixed, suddenly another one is reported)

```suggestion
		require.IsType(t, &sema.InvalidAccessError{}, errs[0])
		assert.Equal(
			t,
			errs[0].(*sema.InvalidAccessError).RestrictingAccess,
			sema.NewEntitlementSetAccess(
				[]*sema.EntitlementType{
					sema.InsertEntitlement,
					sema.MutateEntitlement,
				},
				sema.Disjunction,
			),
		)
		assert.Equal(
			t,
			errs[0].(*sema.InvalidAccessError).PossessedAccess,
			sema.UnauthorizedAccess,
		)
```